source/plugins/go/src/ingestion_token_utils.go

```diff
@@ -442,7 +449,12 @@ func getAgentConfiguration(imdsAccessToken string) (configurationId string, chan
 
 	configurationId = agentConfiguration.Configurations[0].Configurationid
 	channelId = agentConfiguration.Configurations[0].Content.Channels[0].ID
-
+	if len(agentConfiguration.Configurations[0].Content.Extensionconfigurations.Containerinsights) > 0 {
+		// Assuming you have an instance of AgentConfiguration named agentConfiguration
+		ContainerLogSchemaV2 = agentConfiguration.Configurations[0].Content.Extensionconfigurations.Containerinsights[0].Extensionsettings.DataCollectionSettings.EnableContainerLogV2
```

Comment:
ExtensionSettings can be empty and  add  empty/nil check.

Reply:
Extensionsettings and DataCollectionSettings are embedded structs, not pointers to structs or slices/arrays. In Go, struct fields are automatically initialized to their zero values
so if not set, the output of agentConfiguration.Configurations[0].Content.Extensionconfigurations.Containerinsights[0].Extensionsettings will be { { "" "" nil false } }